fission/src/ui/panels/configuring/MatchModeConfigPanel.tsx

```diff
@@ -0,0 +1,294 @@
+                const uniqueConfigsById = Array.from(new Map(combinedConfigs.map(item => [item.id, item])).values())
+
+                setMatchModeConfigs(uniqueConfigsById)
+            } catch (err) {
+                console.error("Error loading match mode configs:", err)
+                globalAddToast?.(
+                    "error",
+                    "Error Loading Match Mode Configs",
+                    "Please check the console for more details"
+                )
+            }
+        }
+
+        loadConfigs()
+    }, [])
+
+    const matchModeConfigElements = useMemo(
+        () =>
+            matchModeConfigs.map(config => {
+                return (
+                    <ItemCard
+                        key={config.id}
+                        id={config.id}
+                        name={config.name || config.id || "Unnamed Match Mode"}
+                        primaryOnClick={() => {
+                            matchConfigSelected(config, openModal)
+                            closePanel("match-mode-config")
+                        }}
+                        secondaryOnClick={
+                            !config.isDefault
+                                ? () => {
+                                      // Delete the config from the local storage
+                                      const updatedConfigs = matchModeConfigs.filter(c => c.id !== config.id)
+                                      setMatchModeConfigs(updatedConfigs)
+                                      // Only save custom configs to local storage
+                                      const customConfigs = updatedConfigs.filter(c => !c.isDefault)
+                                      window.localStorage.setItem("match-mode-configs", JSON.stringify(customConfigs))
+                                      globalAddToast?.(
+                                          "info",
+                                          "Match Mode Config Deleted",
+                                          `Successfully deleted "${config.name}"`
+                                      )
+                                  }
+                                : undefined
+                        }
+                    />
+                )
+            }),
+        [matchModeConfigs, openModal, closePanel]
+    )
+
+    const fileUploadRef = useRef<HTMLInputElement>(null)
+
+    const uploadClicked = () => {
+        if (fileUploadRef.current) {
+            fileUploadRef.current.click()
+        }
+    }
+
+    const validateAndNormalizeMatchModeConfig = (config: unknown): MatchModeConfig | null => {
+        let valid = true
+
+        // Type guard to check if config is an object
+        if (typeof config !== "object" || config === null) {
+            console.error("Match mode config validation failed: config must be an object")
+            globalAddToast?.("error", "Invalid Match Mode Config", "Configuration must be an object")
```

Comment:
since #1204 the optional chaining is not necessary (not just here but whenever you're doing toasts)

```suggestion
            globalAddToast("error", "Invalid Match Mode Config", "Configuration must be an object")
```